kod/object/active/holder/nomoveon/battler/monster/iceper.kod

```diff
@@ -64,19 +100,40 @@ properties:
 
    piAnimation = ANIM_NONE
    piDrawFX = DRAWFX_TRANSLUCENT_50
-   
+
+   prWeaponOverlay = iceperson_sword % sword as default weapon
+
 messages:
 
    Constructed()
    {
-   plResistances = [ [ -99, -ATCK_SPELL_FIRE ],[ 99, -ATCK_SPELL_COLD ]];
-   propagate;
+      plResistances = [ [ -99, -ATCK_SPELL_FIRE ],[ 99, -ATCK_SPELL_COLD ]];
+
+      % Random weapon selection
+      Send(self,@SetWeapon);
+
+      propagate;
+   }
+
+   SetWeapon()
+   "Random chance to spawn a spear variant"
+   {
+      local iRandom;
+
+      iRandom = random(0,1);
+      if iRandom = 1
+      {
+         prWeaponOverlay = iceperson_spear;
+         piBehavior = AI_MOVE_OPTIMAL_RANGE;
+      }
```

Comment:
Should probably also have an else clause the sets the default, otherwise if you call SetWeapon more than once, the variant is no longer random (much more likely to be spear).

Reply:
Oops. Thanks for catching this!